clang/include/clang/Basic/DiagnosticDriverKinds.td

```diff
@@ -288,6 +288,8 @@ def err_drv_expecting_fsycl_with_sycl_opt : Error<
   "The option %0 must be used in conjunction with -fsycl to enable offloading.">;
 def err_drv_fsycl_with_c_type : Error<
   "The option '%0' must not be used in conjunction with '-fsycl', which expects C++ source.">;
+def err_drv_sycl_missing_amdgpu_arch : Error<
+  "Missing AMDGPU architecture for SYCL offloading. Please specify it with -Xsycl-target-backend --offload-arch.">;
```

Comment:
Clang diagnostics do not start with capital letters, and use different punctuations (See https://github.com/intel/llvm/pull/4243 for additional details). Please change the diagnostic to - 

`missing AMDGPU architecture for SYCL offloading; please specify architecture with -Xsycl-target-backend --offload-arch`

@AaronBallman did I get requirements for diagnostics correct?

Reply:
> @AaronBallman did I get requirements for diagnostic correct?

Yes, though I have an additional suggestion here -- you should surround command line arguments with single quotes like you'd do for syntactic elements. e.g.,

`missing AMDGPU architecture for SYCL offloading; specify architecture with '-Xsycl-target-backend --offload-arch'`

(This also removed the "please" for brevity.)

Reply:
I have updated the message as suggested